### ISSUE TITLE
Add Char to Builtins type

### ISSUE THREAD

**Issue by ftomassetti** [LionWeb-io/specification#233]
When working on exporting existing metamodels to LionWeb, one recurring problem I found is with the lack of support for the Char primitive type in the LionCore Builtins, so I think we should add it.

**Comment by dslmeinte** [comment]
The only essential difference between String and Char would be that the latter is an instance of the former with fixed length 1. If we add Char as a primitive type then we'd have to reify that constraint essentially from the serialization level up, right?

**Comment by ftomassetti** [comment]
I would rather treat the two types as independent, as I think in many platforms there is a class representing a single character with its own methods and meaning

**Comment by dslmeinte** [comment]
True, but argument applies for every type that's supported in a platform. Where do we draw the line?

**Comment by ftomassetti** [comment]
I would draw the line to keep in the 30 or so most commonly used concepts that are shared across languages. I would definitely keep String is, as it is probably the single most use class across most languages. I would go as far as to include classes like Date as well

**Comment by joswarmer** [comment]
Date is actually quite complex.

**Comment by dslmeinte** [comment]
Indeed, I wouldn't want to touch dates with a pole, certainly not on the meta-level.

**Comment by enikao** [comment]
char is also not that easy:
* Is this ASCII (7 bit)?
* 8 bit?
   *  Can it be misused as byte?
   *  Which encoding?
* Unicode? 
   * 16/32 bit? 
   * little-endian/big-endian?
* UTF-8? 
   * do we check its a valid UTF-8 character?
* How to handle it on different platforms?

**Comment by enikao** [comment]
In general, I'd keep the builtins _very_ small, and rather offer some other "reusable base" libraries. There it could make sense to e.g. have a `datetime` library, and support it via Yodatime on all applicable platforms.

**Comment by ftomassetti** [comment]
> char is also not that easy:

Doesn't the same apply to strings?

**Comment by dslmeinte** [comment]
Yes, it does, but that's why we specified strings to be Unicode serialized JSON-compliant as UTF-8. I'm with Niko: keep the builtins to the absolute minimum, and let users define their own specific datatypes as they need and see fit.

**Comment by ftomassetti** [comment]
I think that if we indicate that Strings should be in UTF-8, we can apply the same principle to Chars.

Regarding the other types I think it makes sense to provide a separate library outside of the builtins library. However I am interpreting what Niko is writing as that LionWeb will define this separate library, while you are indicating the user itself will define it. In this case I think that there would be value in acting as coordinators and provide a library for very common types, even if outside the builtins library, in order to avoid having too many incompatibilities. If the goal is to support interoperability I think that sharing common types would facilitate that.

**Comment by dslmeinte** [comment]
I think I'm warming up to the idea of a “secondary” stdlib for primitive types, provided we document _extremely clearly_ what its status is in relation to the `builtins` library, and describe what the problems with using specific types (such as dates) would be.

**Comment by enikao** [comment]
## Decision: use secondary library language
* Exact requirements and semantics need to be described in text (as we are programming language-agnostic)
* Don't describe any expectations where to get it or how it is available
* List important adjacent projects (and libraries) somewhere on LW page
